src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java

```diff
@@ -80,6 +93,10 @@ private void clearCommand() {
 
     public void updateCurrentCommandField() {
         currentCommandField.setText(commandManager.getCurrentCommandString());
+
+        Job2dDriver driver = new LineDriverAdapter(drawPanelController,new BasicLine(),"preview");
```

Comment:
the instance of LineDriverAdapter is unnecessarily created many times. Please promote the driver to the rank of the field (choose appropriate name) and construct it next to the initialization of drawPanelController.